app/controllers/posts_controller.rb

```diff
@@ -1,52 +1,51 @@
 class PostsController < ApplicationController
   before_action :authenticate_account!, except:  [ :index, :show ]
-  before_action :set_post, only: [:show, :edit, :update, :destroy]
-  before_action :auth_subscriber, only: [:new]
+  before_action :set_post, only: [:show, :edit, :update, :destroy, :increment_view_count]
+  before_action :find_my_communities, only: [:new, :create, :edit, :update]
   before_action :community_list
   def index
     @posts = Post.all
   end
 
   def show
+    increment_view_count
     @comment = Comment.new
   end
 
   def new
-    @community = Community.find(params[:community_id])
-    @post = Post.new 
+    @post = Post.new
     @drafts = Post.order(created_at: :desc).page(params[:page]).per 5
   end
 
   def create
+    @drafts = Post.all
     @post = Post.new post_values
     @post.account_id = current_account.id
-    @post.community_id = params[:community_id]
+
     if @post.save
       if params[:commit] == "Publish"
         @post.is_drafted = false
         @post.save
         redirect_to community_path(@post.community_id)
-      else 
+      else
         @post.is_drafted = true
         @post.save
         redirect_to draft_path
-      end 
+      end
```

Comment:
this needs to be refactored. I remember @mithun30052001 fixing this. Why is this showing up again? Can you discuss and figure out what went wrong?

Reply:
Resolved it